src/main/kotlin/dev/minn/jda/ktx/JDABuilder.kt

```diff
@@ -0,0 +1,434 @@
+        }
+    
+    var webSocketFactory: WebSocketFactory? = null
+        set(value) {
+            builder.setWebsocketFactory(value)
+            field = value
+        }
+    
+    var rateLimitPool: ScheduledExecutorService? = null
+        set(value) {
+            builder.setRateLimitPool(value, true)
+            field = value
+        }
+    
+    var gatewayPool: ScheduledExecutorService? = null
+        set(value) {
+            builder.setGatewayPool(value, true)
+            field = value
+        }
+    
+    var callbackPool: ExecutorService? = null
+        set(value) {
+            builder.setCallbackPool(value, true)
+            field = value
+        }
+    
+    var eventPool: ExecutorService? = null
+        set(value) {
+            builder.setEventPool(value, true)
+            field = value
+        }
+    
+    var audioPool: ScheduledExecutorService? = null
+        set(value) {
+            builder.setAudioPool(value, true)
+            field = value
+        }
+    
+    var bulkDeleteSplitting: Boolean = true
+        set(value) {
+            builder.setBulkDeleteSplittingEnabled(value)
+            field = value
+        }
+    
+    var enableShutdownHook: Boolean = true
+        set(value) {
+            builder.setEnableShutdownHook(value)
+            field = value
+        }
+    
+    var autoReconnect: Boolean = true
+        set(value) {
+            builder.setAutoReconnect(value)
+            field = value
+        }
+    
+    var eventManager: IEventManager? = null
+        set(value) {
+            builder.setEventManager(value)
+            field = value
+        }
+    
+    var audioSendFactory: IAudioSendFactory? = null
+        set(value) {
+            builder.setAudioSendFactory(value)
+            field = value
+        }
+    
+    var idle: Boolean = false
+        set(value) {
+            builder.setIdle(value)
+            field = value
+        }
+    
+    var activity: Activity? = null
+        set(value) {
+            builder.setActivity(value)
+            field = value
+        }
+    
+    var status: OnlineStatus? = null
+        set(value) {
+            if (value != null) {
+                builder.setStatus(value)
+            }
+            field = value
+        }
+    
+    val eventListeners: MutableCollection<Any> = DelegatingCollection(
+            adder = { item -> builder.addEventListeners(item) },
+            remover = { item -> builder.removeEventListeners(item) },
+                                                              )
+    
+    var maxReconnectDelay: Int = 900
+        set(value) {
+            builder.setMaxReconnectDelay(value)
+            field = value
+        }
+    
+    var sharding: Pair<Int, Int>? = null
+        set(value) {
+            value?.let { builder.useSharding(value.first, it.second) }
+            field = value
+        }
+    
+    fun sharding(shardId: Int, shardTotal: Int) = builder.useSharding(shardId, shardTotal)
```

Comment:
I think you should remove the property `sharding = 1 to 2` doesn't look very nice regardless. You aren't updating the sharding field here either, better just remove it entirely.

Reply:
I agree.